clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -51,12 +52,22 @@ class AzureNetworkClient extends AzureBaseClient {
     def result = new ArrayList<AzureLoadBalancerDescription>()
 
     try {
-      this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers().each {item ->
+      def loadBalancers = this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().listAll().getLoadBalancers()
```

Comment:
can you point to on such case? If it's related to providerCache content, it's because it holds all the cache corresponding to the respective agent and working with a smaller/filtered list is a lot faster.